chromadb/api/client.py

```diff
@@ -56,13 +57,28 @@ def __init__(
         super().__init__(settings=settings)
         self.tenant = tenant
         self.database = database
-        # Create an admin client for verifying that databases and tenants exist
-        self._admin_client = AdminClient.from_system(self._system)
-        self._validate_tenant_database(tenant=tenant, database=database)
 
         # Get the root system component we want to interact with
         self._server = self._system.instance(ServerAPI)
 
+        user_identity = self.resolve_tenant_and_databases()
+
+        maybe_tenant, maybe_database = SharedSystemClient.maybe_set_tenant_and_database(
+            settings.chroma_overwrite_singleton_tenant_database_access_from_auth,
```

Comment:
I am feeling like this overwrite behavior and the consumption of the setting may be cleaner if in the auth module as a implemented base method.

Reply:
See above—I don't have a super strong preference on this so will defer to you if you do, but to me it feels like auth is a server concern and accessing it from our clients feels off. To me, clients should be a thin layer only used to access our server in a clean way. All the implementation logic should be abstracted away. But again, soft preference.